src/MainWindow.cpp

```diff
@@ -1035,44 +1035,52 @@ void MainWindow::executeQuery()
     QString query;
     int execution_start_line = 0;
     int execution_start_index = 0;
+    int execution_start_position = 0;
+    SqlTextEdit *editor = sqlWidget->getEditor();
     if(sender()->objectName() == "actionSqlExecuteLine")
     {
         int cursor_line, cursor_index;
-        SqlTextEdit *editor = sqlWidget->getEditor();
 
         editor->getCursorPosition(&cursor_line, &cursor_index);
 
         execution_start_line = cursor_line;
 
-        int lineStartCursorPosition = editor->positionFromLineIndex(cursor_line, 0);
+        execution_start_position = editor->positionFromLineIndex(cursor_line, 0);
 
         QString entireSQL = editor->text();
-        QString firstPartEntireSQL = entireSQL.left(lineStartCursorPosition);
-        QString secondPartEntireSQL = entireSQL.right(entireSQL.length() - lineStartCursorPosition);
+        QString firstPartEntireSQL = entireSQL.left(execution_start_position);
+        QString secondPartEntireSQL = entireSQL.right(entireSQL.length() - execution_start_position);
 
         QString firstPartSQL = firstPartEntireSQL.split(";").last();
         QString lastPartSQL = secondPartEntireSQL.split(";").first();
 
         query = firstPartSQL + lastPartSQL;
+        db.logSQL(tr("-- EXECUTING LINE AT '%1'\n--").arg(sqlWidget->fileName()), kLogMsg_User);
```

Comment:
I think 'in' sounds a bit better here than 'at'. But I'm not a native speaker either, so who knows :wink:

Also I would suggest replacing
```c++
sqlWidget->fileName()
```
by something like
```c++
ui->tabSqlAreas->tabText(ui->tabSqlAreas->currentIndex()).remove('&')
```
Otherwise the name will always be empty unless you save your statements to a file. That's especially a problem for people who rename their tabs manually and save them all to a project file.

Reply:
Both 'in' and 'at' translate in Spanish usually to 'en' so we always have problems with that :smile:

Much better with the tab name, of course.